Documentation/ExtensionArchitecture/Extbase/Reference/Domain/Model.rst

```diff
@@ -165,16 +173,142 @@ to make a query in the PostRepository taking one or both relationships (first
+:php:`\MyVendor\MyExtension\Domain\Model\Blog`.
+
+Now, also imagine there is a domain rule which states, that all blogs must have
+a title. This rule can easily be followed by letting the blog class have a
+constructor with a required argument :php:`string $title`.
+
+..  literalinclude:: _Hydrating/_Blog1.php
+    :caption: EXT:my_extension/Classes/Domain/Model/Blog.php
+
+This example also shows how the `posts` property is initialized. It is done in
+the constructor because PHP does not allow setting a default value that is of
+type object.
+
+Hydrating objects with constructor arguments
+--------------------------------------------
+
+Whenever the user creates new blog objects in extension code, the aforementioned
+domain rule is followed. It is also possible to work on the :php:`posts`
+:php:`ObjectStorage` without further initialization. :php:`new Blog('title')` is
+all one need to create a blog object with a valid state.
+
+What happens in the :php:`DataMapper` however, is a totally different thing.
+When hydrating an object, the :php:`DataMapper` cannot follow any domain rules.
+Its only job is to map the raw database values onto a :php:`Blog` instance. The
+:php:`DataMapper` could of course detect constructor arguments and try to guess
+which argument corresponds to what property, but only if there is an easy
+mapping, that means, if the constructor takes the argument :php:`string $title`
+and updates the property `title` with it.
+
+To avoid possible errors due to guessing, the :php:`DataMapper` simply ignores
+the constructor at all. It does so with the help of the library
+`doctrine/instantiator`_.
+
+But there is more to all this.
+
+..  _doctrine/instantiator: https://github.com/doctrine/instantiator
+
+Initializing objects
+--------------------
+
+Have a look at the :php:`$posts` property in the example above. If the
+:php:`DataMapper` ignores the constructor, that property is in an invalid state,
+that means, uninitialized.
+
+To address this problem and possible others, the :php:`DataMapper` will call the
+method :php:`initializeObject(): void` on models, if it exists.
+
+Here is an updated version of the model:
+
+..  literalinclude:: _Hydrating/_Blog2.php
+    :caption: EXT:my_extension/Classes/Domain/Model/Blog.php
+
+This example demonstrates how Extbase expects the user to set up their models.
+If the method :php:`initializeObject()` is used for initialization logic that
+needs to be triggered on initial creation **and** on hydration. Please mind
+that :php:`__construct()` **should** call :php:`initializeObject()`.
+
+If there are no domain rules to follow, the recommended way to set up a model
+would then still be to define a :php:`__construct()` and
+:php:`initializeObject()` method like this:
+
+..  literalinclude:: _Hydrating/_Blog3.php
+    :caption: EXT:my_extension/Classes/Domain/Model/Blog.php
+
+Mutating objects
+----------------
+
+Some few more words on mutators (setter, adder, etc.). One might think that
```

Comment:
I think there should be a short paragraph for explaining mutators, as it might not be known to everyone.
Also about the difference of simple properties where getX and setX is enough, and others that use addX removeX additional.

Reply:
It is explained: mutators (setter, adder, etc.) Don't know what there should be additionally explained. Also, this is not in scope of this PR. If there is the necessity for explaining, this can be done in a separate PR.